### ISSUE TITLE
404 link on template_file documentation

### ISSUE THREAD

**Issue by ricksebak** [hashicorp/terraform#20055]
I'm not fully sure if this is even the right place to report documentation bugs, but if you go to https://www.terraform.io/docs/providers/template/d/file.html and ctrl-f for "Terraform template syntax", you'll see two links for that phrase, and both are 404'ed.

Not a huge deal or anything, but I figured I would point it out to the powers that be.

**Comment by mildwonkey** [comment]
Thanks @ricksebak , that is good to know! 

I found a third broken link on that same page:
https://www.terraform.io/docs/configuration/expressions.html#string-templates

**Comment by apparentlymart** [comment]
Hi @ricksebak! Thanks for reporting this.

I thought I'd already replied here but I guess I must've closed the tab without hitting the send button :roll_eyes: 

This problem has arisen from the template provider with Terraform 0.12.0 support being released before Terraform 0.12.0 itself, and so it's anticipating some new documentation that won't be published until Terraform Core v0.12.0 is final. We're gonna put some temporary measures in place to make these lead _somewhere_ for now, but ultimately these links should work properly once the site is updated as part of the v0.12.0 final release.

In the mean time, you can find the unpublished draft of the docs this is trying to link to here: [String Templates](https://github.com/hashicorp/terraform/blob/db36ccb316ea55fee5632e8a555e50d9731658f3/website/docs/configuration/expressions.html.md#string-templates). There will be some broken links in that page too because the markdown here is set up with links that make sense only when published on our website, but the content itself should be readable there.

**Comment by ricksebak** [comment]
@apparentlymart , makes sense. If it's a known issue, or if this GH issue is otherwise not needed, feel free to just close this. Thanks all.
